catkit/emulators/iris_ao_controller.py

```diff
@@ -0,0 +1,114 @@
+import astropy.units as u
+import poppy
+
+from catkit.hardware import testbed_state
+import catkit.hardware.iris_ao.iris_ao_controller
+import catkit.hardware.iris_ao.segmented_dm_command as segmented_dm_command
+import catkit.hardware.iris_ao.util
+from catkit.interfaces.Instrument import SimInstrument
+
+
+class PoppyIrisAODM(poppy.dms.HexSegmentedDeformableMirror):
+
+    @property
+    def number_of_segments(self):
+        return self.dm_shape
+
+    def __init__(self, mcf_filename, custom_flat_filename, **super_kwargs):
+        super().__init__(**super_kwargs)
+
+        self.mirror_serial = None
+
+        # Read the manufacturers .mcf "flat" file.
+        # This is implicitly applied by the driver/controller and is NOT a present contribution of the command sent to
+        # the driver/controller by catkit.hardware.iris_ao.iris_ao_controller.IrisAoDmController.
+        self.mcf_filename = mcf_filename
+        #mcf_data, self.mirror_serial = catkit.hardware.iris_ao.util.read_mcf(self.mcf_filename, self.number_of_segments)
+
+        self.custom_flat_filename = custom_flat_filename
+        custom_flat_data = catkit.hardware.iris_ao.util.read_ini(self.custom_flat_filename, self.number_of_segments)   # this returns a DM command dict
+
+        # A flat Poppy surfaces := 0.
+        self.relaxed_poppy_surface = self.invert_data(custom_flat_data)  # + mcf_data
+        self.relax()  # ??? See https://github.com/spacetelescope/catkit/issues/63 (we don't currently relax the bostons like this).
+
+    def actuate(self, data):
```

Comment:
How about we call this `set_surface()`, like for the Bostons?

Reply:
Lol I did originally and even conditional called ``super().set_surface()`` for non-dicts, but there's no base ``set_surface()`` for the segmented dms like there is for continuous. ``set_surface()`` belongs to poppy. So I wondered if it didn't make sense to do so. 🤷 

Happy to change it, again, though.

Reply:
You're exactly right ^^ I was talking to Marshall and I considered adding a base `set_surface()` for the hex DM in `poppy` but there's really no gain from it. Let's still change this one to `set_surface()` as it makes it more obvious what it is doing, if you're ok with that.